### PR TITLE
Fix display version in OCI image

### LINKED ISSUE #257: OCI image created with ko in a release misses version informations
## In what area(s)?

Release 

## What version of Knative Client?

```
Version:      v20190710-local-31dc57a-dirty
Build Date:   2019-07-10 09:51:46
Git Revision: 31dc57a
Dependencies:
- serving:    v0.6.0
```

## What version of Knative Serving running on your cluster?

0.7.x

## Expected Behavior

The image created during the release in https://github.com/knative/client/blob/e1614f95e662e9e6f7f520eb2b33181d7b264a28/hack/release.sh#L36 with `ko` should also contain all version informations (propagated via `ldflags`)

## Actual Behavior

Running the created Docker image with the coordinates from the 0.2.0 release  yields:

```
docker run --rm -it gcr.io/knative-releases/github.com/knative/client/cmd/kn@sha256:393035ce99f1e639ba859d35ffb8b3f328722a2f22ab4b4ea1d889b9c015af49 version

Unable to find image 'gcr.io/knative-releases/github.com/knative/client/cmd/kn@sha256:393035ce99f1e639ba859d35ffb8b3f328722a2f22ab4b4ea1d889b9c015af49' locally
sha256:393035ce99f1e639ba859d35ffb8b3f328722a2f22ab4b4ea1d889b9c015af49: Pulling from knative-releases/github.com/knative/client/cmd/kn
155814304360: Pull complete
4003b5b92ca9: Pull complete
1f704ff7c739: Pull complete
Digest: sha256:393035ce99f1e639ba859d35ffb8b3f328722a2f22ab4b4ea1d889b9c015af49
Status: Downloaded newer image for gcr.io/knative-releases/github.com/knative/client/cmd/kn@sha256:393035ce99f1e639ba859d35ffb8b3f328722a2f22ab4b4ea1d889b9c015af49
Version:
Build Date:
Git Revision:
Dependencies:
- serving:
```

### PR DESCRIPTION
**Pull request by dsimansk**
## Changes

* Fix display version in OCI image

## Reference

Fixes #257 

/cc @rhuss @vyasgun

### CODE REVIEW

File: .ko.yaml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 defaultBaseImage: gcr.io/distroless/static:nonroot
 baseImageOverrides:
   knative.dev/client/cmd/kn: docker.io/library/alpine:latest
+builds:
+  - id: knative.dev/client/cmd/kn
+    main: ./cmd/kn
+    ldflags:
+      - -X knative.dev/client/pkg/kn/commands/version.Version={{.Env.KN_BUILD_VERSION}}
+      - -X knative.dev/client/pkg/kn/commands/version.GitRevision={{.Env.KN_BUILD_GITREV}}
+      - -X 'knative.dev/client/pkg/kn/commands/version.BuildDate={{.Env.KN_BUILD_DATE}}'
```
Comment (dsimansk) [inline .ko.yaml:10]: @rhuss that looks much better, even with global env vars. :) At least it's clear what we're trying to achieve here and without in-place manipulation of files.

Comment (dsimansk) [reply .ko.yaml:10]:
```console
➜  client git:(pr/fix-257) echo $KN_BUILD_VERSION $KN_BUILD_GITREV $KN_BUILD_DATE    
1.4.0 603f3f59f 2022-04-26 15:49:21
➜  client git:(pr/fix-257) podman run --rm -it ko.local/kn-0eb649368dcc185030f663a9db12e136:53980ed270a73c5b5a40e1f8d249823205e6c3705a3bea5b413691a98ff013e3 version
Version:      1.4.0
Build Date:   2022-04-26 15:49:21
Git Revision: 603f3f59f
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.31.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.31.0)
  - eventing.knative.dev/v1 (knative-eventing v0.31.0)
```

Comment (rhuss) [reply .ko.yaml:10]: thanks, looks great !

/approve
/lgtm

**Pull request merged**